tf-modules/archive/fetch_or_create_rsa_keys.sh

```diff
@@ -25,9 +21,6 @@ if ! aws s3api head-object --bucket "$BUCKET" --key "$PRIVATE_KEY" >/dev/null 2>
 fi
 
 echo "Uploading public key for ${STACK}"
-# Create and upload public key to S3
-if ! aws s3api head-object --bucket "$BUCKET" --key "$PRIVATE_KEY" >/dev/null 2>&1; then
-  aws s3 cp "s3://${BUCKET}/${PRIVATE_KEY}" - 2>/dev/null |\
-    openssl rsa -in /dev/stdin -outform PEM -pubout -out /dev/stdout 2>&1 |\
-    aws s3 cp - "s3://${BUCKET}/${PUBLIC_KEY}" >/dev/null 2>&1
-fi
+aws s3 cp "s3://${BUCKET}/${PRIVATE_KEY}" - 2>/dev/null |\
```

Comment:
Why not check if the private key exists before trying this? Also, i don't understand how the public key is being generated. Is it using openssl to generate the public key from the private key?

Reply:
I'm guessing you meant, "why not check if the _public_ key exists ..." I did it this way because I wanted to make sure that the public key always matches the private key. There could potentially be a case where a new private key is created but, because the previous public key was already in S3, it wouldn't create a public key that matched the private key.

To answer your second question, yes, this is just generating the public key from the private key.